TTS/trainer.py

```diff
@@ -640,10 +651,24 @@ def _optimize(
                 update_lr_scheduler = scale_prev <= scaler.get_scale()
                 loss_dict["amp_scaler"] = scaler.get_scale()  # for logging
         else:
+            if hasattr(self.model, "capacitron_vae"):
+                loss_dict["capacitron_vae_beta_loss"].backward()
+                capacitron_SGD_optimizer.step()
+                capacitron_SGD_optimizer.zero_grad()
+                optimizer.zero_grad()
+                # Gradient clipping below needs to exclude capacitron_beta
+                model_params_to_clip = []
+                for name, param in model.named_parameters():
+                    if param.requires_grad:
+                        if name != "capacitron_vae_layer.beta":
+                            model_params_to_clip.append(param)
+            else:
+                model_params_to_clip = model.parameters()
+
```

Comment:
The same here. These lines should leave the trainer .

Reply:
I'm not sure where to put this logic with the optimizer steps and zero grads

Reply:
can't you type a new optimizer class that uses these two optimizers internally?

Reply:
I didn't think of that, I'll give it a try, thanks!